### DISCUSSION TITLE
scheduled job to upgrade all workers?

### DISCUSSION THREAD

**Discussion by nycalex** [jhuckaby/Cronicle#730] [General]
good morning

i was thinking... what if i create a on-demand (manual execution) job to run as root a script "/opt/cronicle/bin/control.sh upgrade" on all worker nodes?

and run it detached, to be safe...

Would it work or a bad idea?

Thanks!!!

**Comment by jhuckaby** [comment]
LOL!  That's funny.  

I suppose if you ran it detached it might work, but I've never tried such a thing before.  There may be unforeseen issues.

I definitely wouldn't attempt this on a production cluster.

**Reply by nycalex** [reply]
What are the possible reasons it's not a production-grade idea? 

An ability to upgrade an entire cluster is a huge plus

**Reply by jhuckaby** [reply]
I dunno, I just haven't tried it before.  I honestly don't know what will happen.  Maybe nothing?  Maybe it'll work great?

The risk is that it's updating itself from inside itself, while it is running a job, itself.  It just seems like asking for trouble, is all.

Do you have some test servers you can try it on, that aren't running your live production jobs?

**Reply by nycalex** [reply]
I am just playing in a vm environment, i can try next time there's an update available.

**Reply by jhuckaby** [reply]
Cool!

> An ability to upgrade an entire cluster is a huge plus

I totally agree this would be a great built-in feature.  I'm working on something like this for v2 (due out later this year).
